commit 49edfa3f324c1dbff482b1e141f0284121031984
Author: ICHINOSE Shogo <shogo82148@gmail.com>
Date:   Fri Oct 13 07:21:51 2023 +0900

    Revert "read the version numbers rather than generating version.h"
    
    This reverts commit aae1287098c38fb615e6f7761b04d6a8b284cacd.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e3df812ea..87efe7824 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,6 +25,17 @@ CMAKE_POLICY(SET CMP0042 NEW)
 
 PROJECT(h2o)
 
+SET(VERSION_MAJOR "2")
+SET(VERSION_MINOR "3")
+SET(VERSION_PATCH "0")
+SET(VERSION_PRERELEASE "-DEV")
+SET(VERSION "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}${VERSION_PRERELEASE}")
+SET(LIBRARY_VERSION_MAJOR "0")
+SET(LIBRARY_VERSION_MINOR "16")
+SET(LIBRARY_VERSION_PATCH "0")
+SET(LIBRARY_VERSION "${LIBRARY_VERSION_MAJOR}.${LIBRARY_VERSION_MINOR}.${LIBRARY_VERSION_PATCH}${VERSION_PRERELEASE}")
+SET(LIBRARY_SOVERSION "${LIBRARY_VERSION_MAJOR}.${LIBRARY_VERSION_MINOR}")
+
 INCLUDE(GNUInstallDirs)
 INCLUDE(CheckCSourceCompiles)
 INCLUDE(CMakePushCheckState)
@@ -32,6 +43,7 @@ INCLUDE(ExternalProject)
 INCLUDE(deps/picotls/cmake/dtrace-utils.cmake)
 INCLUDE(deps/picotls/cmake/boringssl-adjust.cmake)
 
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/version.h.in ${CMAKE_CURRENT_SOURCE_DIR}/include/h2o/version.h)
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/libh2o.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libh2o.pc @ONLY)
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/libh2o-evloop.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libh2o-evloop.pc @ONLY)
 
@@ -615,20 +627,6 @@ ELSE ()
     ADD_LIBRARY(libh2o-evloop ${LIB_SOURCE_FILES})
 ENDIF (WSLAY_FOUND)
 
-EXECUTE_PROCESS(
-    COMMAND sed -n -E "/#define {1,}H2O_VERSION_BASE .*-DEV\"$/s//YES/p" ${CMAKE_CURRENT_SOURCE_DIR}/include/h2o/version.h
-    OUTPUT_VARIABLE IS_PRERELEASE
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-EXECUTE_PROCESS(
-    COMMAND sed -n -E "/#define {1,}H2O_LIBRARY_VERSION {1,}\"(.*)\"$/s//\\1/p" ${CMAKE_CURRENT_SOURCE_DIR}/include/h2o/version.h
-    OUTPUT_VARIABLE LIBRARY_VERSION
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-EXECUTE_PROCESS(
-    COMMAND sed -n -E "/#define {1,}H2O_LIBRARY_VERSION {1,}\"(.*)\\.[0-9]{1,}\"$/s//\\1/p" ${CMAKE_CURRENT_SOURCE_DIR}/include/h2o/version.h
-    OUTPUT_VARIABLE LIBRARY_SOVERSION
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-MESSAGE(STATUS "extracted version numbers from ${CMAKE_CURRENT_SOURCE_DIR}/include/h2o/version.h:\n\tprerelease=${IS_PRERELEASE}\n\tlibrary version=${LIBRARY_VERSION}\n\tlibrary soversion=${LIBRARY_SOVERSION}")
-
 SET_TARGET_PROPERTIES(libh2o PROPERTIES
     OUTPUT_NAME h2o
     VERSION ${LIBRARY_VERSION}
diff --git a/include/h2o/version.h b/include/h2o/version.h
index c22a22a24..da6382591 100644
--- a/include/h2o/version.h
+++ b/include/h2o/version.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2014-2023 DeNA Co., Ltd., Kazuho Oku, Fastly
+ * Copyright (c) 2014,2015 DeNA Co., Ltd.
  *
  * Permission is hereby granted, free of charge, to any person obtaining a copy
  * of this software and associated documentation files (the "Software"), to
@@ -31,14 +31,6 @@
 #define H2O_VERSION_MINOR 3
 #define H2O_VERSION_PATCH 0
 
-#ifdef H2O_GITREV
-#define H2O_VERSION H2O_VERSION_BASE "@" H2O_TO_STR(H2O_GITREV)
-#else
-#define H2O_VERSION H2O_VERSION_BASE
-#endif
-
-/* `H2O_LIBRARY_VERSION` is a hard-coded string with three digits, that's the format we parse in CMakeLists.txt */
-#define H2O_LIBRARY_VERSION "0.16.0"
 #define H2O_LIBRARY_VERSION_MAJOR 0
 #define H2O_LIBRARY_VERSION_MINOR 16
 #define H2O_LIBRARY_VERSION_PATCH 0
diff --git a/version.h.in b/version.h.in
new file mode 100644
index 000000000..3d5cd9801
--- /dev/null
+++ b/version.h.in
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2014,2015 DeNA Co., Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+#ifndef h2o__version_h
+#define h2o__version_h
+
+#ifdef H2O_HAS_GITREV_H
+#include "h2o/gitrev.h"
+#endif
+
+#ifdef H2O_GITREV
+#define H2O_VERSION "@VERSION@@" H2O_TO_STR(H2O_GITREV)
+#else
+#define H2O_VERSION "@VERSION@"
+#endif
+
+#define H2O_VERSION_MAJOR @VERSION_MAJOR@
+#define H2O_VERSION_MINOR @VERSION_MINOR@
+#define H2O_VERSION_PATCH @VERSION_PATCH@
+
+#define H2O_LIBRARY_VERSION_MAJOR @LIBRARY_VERSION_MAJOR@
+#define H2O_LIBRARY_VERSION_MINOR @LIBRARY_VERSION_MINOR@
+#define H2O_LIBRARY_VERSION_PATCH @LIBRARY_VERSION_PATCH@
+
+#endif
